File: 2391.cpp
struct Solution {
    int garbageCollection(const vector<string>& garbage, const vector<int>& travel) {
        int res = 0, travelTime = 0, lastTravelTimes[3] = {0, 0, 0};
        for (int i = 0; i < garbage.size(); ++i){
            const string &house = garbage[i];
            res += house.size();
            if (house.find('G') != string::npos){
                lastTravelTimes[0] = travelTime;
            }
            if (house.find('M') != string::npos){
                lastTravelTimes[1] = travelTime;
            }
            if (house.find('P') != string::npos){
                lastTravelTimes[2] = travelTime;
            }
            travelTime += (i < travel.size() ? travel[i] : 0);
        }
        return res + lastTravelTimes[0] + lastTravelTimes[1] + lastTravelTimes[2];
    }
};

File: sol/2391.py
class Solution:
    def garbageCollection(self, garbage: List[str], travel: List[int]) -> int:
        res = 0
        travelTime = 0
        lastTravelTimes = [0] * 3
        # keys = 'GMP'
        for i, house in enumerate(garbage):
            res += len(house)
            # for idx, sig in enumerate(keys):
            #     if sig in house:
            #         lastTravelTimes[idx] = travelTime
            if 'G' in house:
                lastTravelTimes[0] = travelTime
            if 'M' in house:
                lastTravelTimes[1] = travelTime
            if 'P' in house:
                lastTravelTimes[2] = travelTime
            travelTime += travel[i] if i < len(travel) else 0
        return res + sum(lastTravelTimes)
